treetraversal.py

```
@@ -0,0 +1,129 @@
+
+class WidgetTypes:
+    COMPONENT = "component"
+    NODE = "node"
+    UNFORMATTED_TEXT = "unformatted-text"
+    BUTTON = "button"
+
+@dataclass
+class WidgetNode:
+    type: WidgetTypes
+    props: BehaviorSubject[Dict[str, any]] = field(default_factory=lambda: BehaviorSubject({}))
+    children: BehaviorSubject[List[Renderable]] = field(default_factory=lambda: BehaviorSubject([]))
```

Ruff (0.8.2)

Undefined name `Renderable`

(F821)